doc_source/falco-fargate-ptrace-security.md

```diff
@@ -0,0 +1,119 @@
+# Securing Fargate Tasks with Falco Runtime Security
+
+ECS Fargate `1.4.0` [announced support](https://aws.amazon.com/about-aws/whats-new/2020/04/aws-fargate-launches-platform-version-14/) for the `CAP_SYS_PTRACE` linux capability.
+
+This new feature can be implemented with [Falco](falco.org), an open source runtime security project originally built by Sysdig, Inc and later donated to the Cloud Native Computing Foundation.
+
+Falco uses `ptrace(2)` with Fargate in ECS to detect anomalous behavior at runtime. 
+
+Falco can be configured to send alerts to STDOUT such that ECS can consume the logs. They can then be used to trigger alerts and alarms using AWS logging solutions such as CloudWatch. 
+
+## Dependencies
+
+In order to begin securing a task in ECS a few components need to be installed within the container.
+
+ - Falco userspace daemon
+ - Falco pdig tracing utility
+ 
+After these components are installed in the container, Falco can be used in the following ways.
+
+ - Launch an arbitrary process, and begin tracing the original process and all of it's child processes. 
+ - Attach to a running process, and begin tracing the process and all of it's child processes. 
```

Comment:
```suggestion
 - Launch an arbitrary process, and begin tracing the original process and all of its child processes. 
 - Attach to a running process, and begin tracing the process and all of its child processes. 
```

(attaching to multiple threads is not implemented yet, so the second bullet is ~half true for now)

Reply:
Can we open a feature for this in the Falco `pdig` repository? Can you also suggest some language that would be more appropriate here?

Reply:
PS: Thank you for the typo suggestion - will accept once we figure out the threading things :heart:

Reply:
We need to implement attaching to multiple threads in pdig anyway so feel free to open an issue for this :)

I really hope the limitation is temporary, but currently the second bullet should be more like:

> Attach to a running thread, and begin tracing that thread and all of its future child threads and processes

(there's no such thing as a child thread but this isn't really the place to dig into this)